### PR TITLE
[sdk] Bump to core v10.0.0-beta.21

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Bump to core v10.0.0-beta.21.</changelog>`.

### Summary of changes

Bump gl-native dependency to v10.0.0-beta.21 and common dependency to 11.0.2 and fix breaking upstream changes.

### User impact (optional)

- Refactor CameraOptions and change Map#getCameraState method to return non-nullable CameraState record
- Remove Map#getMinZoom, Map#getMaxZoom and Map#getScale methods that are duplicate of functionality provided by Map#getBounds and Map#getCameraState methods. Map#getBounds returns new CameraBounds type with non-nullable fields.
- Remove Map#setDefaultFramebufferObject
- Remove Map#dumpDebugLog
- Remove isPanning, isRotating, isScaling and cancelTransitions methods in Map API
- Remove following methods from `MapCameraManagerDelegate` (formly `MapCameraDelegate`): `getLan()`, `getLon()`, `getPitch()`, `getBearing()`, `getPadding()`, `setBearing(double)`. Those properties could be accessed now from `MapCameraManagerDelegate.cameraState` directly.

### CODE REVIEW

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt [hunk 1]
```diff
@@ -194,26 +185,21 @@ class CameraAnimatorsFactory internal constructor(mapDelegateProvider: MapDelega
...
-        return Array(1) {
-          CameraCenterAnimator(
-            options = cameraAnimatorOptions(target) {
-              startValue(start)
-            },
-            block = defaultAnimationParameters[CameraAnimatorType.CENTER]
-          )
-        }
-      }
+    return Array(1) {
```
Comment (tobrun) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt:195]: not sure why this syntax is preferred over arrayOf? Because this creates a fixed size array?

File: plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImplTest.kt [hunk 2]
```diff
@@ -979,6 +990,16 @@ class CameraAnimationsPluginImplTest {
...
+  private fun CameraOptions.toCameraState(): CameraState {
+    return CameraState(
+      center ?: cameraOptions.center!!,
+      padding ?: cameraOptions.padding!!,
+      zoom ?: cameraOptions.zoom!!,
+      bearing ?: cameraOptions.bearing!!,
+      pitch ?: cameraOptions.pitch!!
+    )
+  }
+
```
Comment (tobrun) [inline plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImplTest.kt:1002]: Would this extension function be worth exposing publicly?

File: sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt [hunk 3]
```diff
@@ -220,17 +214,18 @@ interface MapCameraManagerDelegate {
...
   fun setCamera(cameraOptions: CameraOptions)
 
   /**
-   * Returns the current camera options
+   * Returns the current camera state
    *
-   * @param padding Optional edge padding
    */
-  fun getCameraOptions(padding: EdgeInsets?): CameraOptions
+  fun getCameraState(): CameraState
```
Comment (pengdev) [inline sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt:228]: can we use a `val cameraState: CameraState` here? So that users would just use `mapboxMap.cameraState.bearing` for example?

File: plugin-compass/src/test/java/com/mapbox/maps/plugin/compass/CompassViewPluginTest.kt [hunk 4]
```diff
@@ -359,8 +351,7 @@ class CompassViewPluginTest {
       0.0,
       0.0,
       -0.23,
-      null,
-      null
+      arrayOf(0.0, 0.0, 0.0, 0.0)
```
Comment (tobrun) [inline plugin-compass/src/test/java/com/mapbox/maps/plugin/compass/CompassViewPluginTest.kt:354]: could we leverage a default value instead?

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 5]
```diff
@@ -345,21 +346,14 @@ class MapboxMap internal constructor(
...
    */
-  override fun getMaxZoom() = nativeMapWeakRef.call { this.maxZoom }
-
-  /**
-   * The current zoom factor applied on the map meaning 2 ^ Current zoom
-   *
-   * @return Scale value.
-   */
-  override fun getScale() = nativeMapWeakRef.call { this.scale }
+  override fun getMaxZoom() = nativeMapWeakRef.call { this.bounds.maxZoom }
```
Comment (tobrun) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:356]: Upstream we made the decision to not expose getMinZoom and getMaxZoom directly on map object, only through bounds API. Shouldn't we do the same here?

**Review by tobrun — APPROVED** [review]
Some follow-up questions below

File: sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt (hunk 3 above)
Comment (kiryldz) [reply sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt:228]: We can do it and also remove all the methods here like `getLan`, `getLon` etc. I believe they were introduced to avoid `!!` explicit casting. I also would advise removing `setBearing` method as it looks a bit strange all alone.
@mapbox/maps-android WDYT?

Comment (pengdev) [reply sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt:228]: sounds good to me

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt (hunk 5 above)
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:356]: I guess you're correct, will do it! I basically left them because we use those methods for camera animations but API alignment is much more important of course.

File: plugin-compass/src/test/java/com/mapbox/maps/plugin/compass/CompassViewPluginTest.kt (hunk 4 above)
Comment (kiryldz) [reply plugin-compass/src/test/java/com/mapbox/maps/plugin/compass/CompassViewPluginTest.kt:354]: Sure

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt (hunk 1 above)
Comment (kiryldz) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt:195]: IDK 😄 I may change it but don't think it actually matters that much

File: app/src/main/java/com/mapbox/maps/testapp/examples/RestrictBoundsActivity.kt [hunk 6]
```diff
@@ -130,37 +132,48 @@ class RestrictBoundsActivity : AppCompatActivity() {
...
       )
       .minZoom(2.0)
       .build()
 
-    private val INFINITE_BOUNDS: BoundOptions = BoundOptions.Builder().infiniteBounds(true).build()
+    private val INFINITE_BOUNDS: CameraBoundsOptions = CameraBoundsOptions.Builder()
+      .bounds(
+        CoordinateBounds(
+          Point.fromLngLat(170.0202020, -20.0),
+          Point.fromLngLat(190.0, 20.0),
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/examples/RestrictBoundsActivity.kt:173]: why these two points?

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraTransform.kt [hunk 7]
```diff
@@ -26,11 +26,9 @@ internal object CameraTransform {
 
   fun ScreenCoordinate.offset(arg: ScreenCoordinate) = ScreenCoordinate(x - arg.x, y - arg.y)
 
-  fun getMapCenter(edgeInsets: EdgeInsets?, mapSize: Size): ScreenCoordinate {
-    val centerX = (mapSize.width - (edgeInsets?.left ?: 0.0) - (edgeInsets?.right ?: 0.0)) / 2.0 +
-      (edgeInsets?.left ?: 0.0)
-    val centerY = (mapSize.height - (edgeInsets?.top ?: 0.0) - (edgeInsets?.bottom ?: 0.0)) / 2.0 +
-      (edgeInsets?.top ?: 0.0)
+  fun getMapCenter(edgeInsets: EdgeInsets, mapSize: Size): ScreenCoordinate {
```
Comment (pengdev) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraTransform.kt:29]: why do we change the signature? seems not related to the recent changes? can we make it 
```
fun getMapCenter(edgeInsets: EdgeInsets? = null, mapSize: Size): ScreenCoordinate {
```

File: app/src/main/java/com/mapbox/maps/testapp/examples/RestrictBoundsActivity.kt (hunk 6 above)
Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/examples/RestrictBoundsActivity.kt:173]: as per docs if infinite bounds flag is `true` those values are not taken into account. so just took any values 😄

File: plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImplTest.kt (hunk 2 above)
Comment (pengdev) [reply plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImplTest.kt:1002]: emm. I think this extension function relies on the local variable `cameraOptions`, wouldn't be applicable for universal use cases.

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 8]
```diff
@@ -357,7 +357,7 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
           val scrollDist = event.getAxisValue(MotionEvent.AXIS_VSCROLL)
 
           // Scale the map by the appropriate power of two factor
-          val currentZoom = mapCameraManagerDelegate.getCameraOptions(null).zoom
+          val currentZoom = mapCameraManagerDelegate.cameraState.zoom
           val cachedAnchor = cameraAnimationsPlugin.anchor
           currentZoom?.let {
```
Comment (pengdev) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:362]: nit: doesn't need. `?.let`

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 9]
```diff
@@ -853,7 +853,7 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
     // cameraChangeDispatcher.onCameraMoveStarted(CameraChangeDispatcher.REASON_API_GESTURE);
 
     // Calculate map bearing value
-    val currentBearing = mapCameraManagerDelegate.getCameraOptions(null).bearing
+    val currentBearing = mapCameraManagerDelegate.cameraState.bearing
     rotateCachedAnchor = cameraAnimationsPlugin.anchor
     currentBearing?.let {
```
Comment (pengdev) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:858]: nit: doesn't need. ?.let

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraTransform.kt (hunk 7 above)
Comment (kiryldz) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraTransform.kt:29]: it's related to gl-native - now edge insets are always non-null

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt (hunk 8 above)
Comment (kiryldz) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:362]: nice catch!

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt (hunk 9 above)
Comment (kiryldz) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:858]: nice catch!

File: sdk/src/androidTest/java/com/mapbox/maps/CameraAnimationsPluginTest.kt [hunk 10]
```diff
@@ -95,7 +95,7 @@ class CameraAnimationsPluginTest : BaseAnimationMapTest() {
       assertTrue(animatorListener2.ended)
       assertTrue(animatorListener2.canceled)
 
-      assertEquals(180.0, mapView.getMapboxMap().getCameraOptions(null).bearing!!, EPS)
+      assertEquals(180.0, mapView.getMapboxMap().cameraState.bearing!!, EPS)
```
Comment (pengdev) [inline sdk/src/androidTest/java/com/mapbox/maps/CameraAnimationsPluginTest.kt:98]:
```suggestion
      assertEquals(180.0, mapView.getMapboxMap().cameraState.bearing, EPS)
```

File: sdk/src/androidTest/java/com/mapbox/maps/CameraAnimationsPluginTest.kt [hunk 11]
```diff
@@ -439,7 +439,7 @@ class CameraAnimationsPluginTest : BaseAnimationMapTest() {
     if (pitchListenerThree.latchEnd.await(LATCH_MAX_TIME, TimeUnit.MILLISECONDS)) {
       assertFalse(pitchListenerThree.canceled)
       assertEquals(30.0, currentPitch, EPS)
-      assertEquals(30.0, mapView.getMapboxMap().getCameraOptions(null).pitch!!, EPS)
+      assertEquals(30.0, mapView.getMapboxMap().cameraState.pitch!!, EPS)
```
Comment (pengdev) [inline sdk/src/androidTest/java/com/mapbox/maps/CameraAnimationsPluginTest.kt:442]:
```suggestion
      assertEquals(30.0, mapView.getMapboxMap().cameraState.pitch, EPS)
```

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 12]
```diff
@@ -848,15 +822,6 @@ class MapboxMap internal constructor(
...
 
-  /**
-   * @brief Rebinds the default frame buffer object
-   *
-   * @param framebufferId ID of the new frame buffer object
-   */
-  fun setDefaultFramebufferObject(framebufferId: Int) {
-    nativeMapWeakRef.call { this.setDefaultFramebufferObject(framebufferId) }
-  }
-
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:859]: Is this removed from gl-native?

Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:859]: yes

**Review by pengdev — APPROVED** [review]
LGTM % failing tests.
